### DISCUSSION TITLE
Frequent restarts in scaleset listener in version 0.9.0

### DISCUSSION THREAD

**Discussion by rekha-prakash-maersk** [actions/actions-runner-controller#3560] [Questions] [answered]
Noticing this error in listener app causing the listener to restart very frequently, but not finding anything odd on the controller logs,

`2024/05/31 07:01:59 Application returned an error: failed to handle message: failed to delete message: failed to delete message: {"$id":"1","innerException":null,"message":"The user 'System:PublicAccess;aaaaaaaa-aaaa-aaaa-aaaa-aaaaaaaaaaaa' is not authorized to access this resource.","typeName":"Microsoft.TeamFoundation.Framework.Server.UnauthorizedRequestException, Microsoft.TeamFoundation.Framework.Server","typeKey":"UnauthorizedRequestException","errorCode":0,"eventId":3000}
`

**Comment by nikola-jokic** [comment, marked as answer]
Hey, this issue is related to [this](https://github.com/actions/actions-runner-controller/pull/3529) fix. Basically, if you are unlucky that your session expires when delete message is happening, then we don't refresh the session and the listener crashes.
Please upgrade to 0.9.2.

**Reply by rekha-prakash-maersk** [reply]
@nikola-jokic Thanks for the update. can you please confirm if this upgrade has any CRD changes that needs the CRDS to be deleted before the upgrade.

**Reply by nikola-jokic** [reply]
Patch releases should not have any CRD changes, so it should be okay.
